### PR TITLE
Deployment CLI tests

### PR DESCRIPTION
**Pull request by cicdw**
### Summary

I created some tech debt last week that I want to revisit by adding tests for critical path behavior of the deployment build / apply CLI commands.  This PR introduces a basic setup for those tests that I wanted to get reviewed before I go too far.

### CODE REVIEW

File: tests/cli/test_deployment_build.py [hunk 1]
```diff
@@ -1,12 +1,50 @@
...
+            ["deployment", "build", "./dog.py"],
+            expected_output_contains=["A name for this deployment must be provided"],
+            expected_code=1,
+        )
+
+
+class TestManifestGeneration:
+    @pytest.fixture
+    def cwd(self, tmp_path):
+        """Uses a local file system method to move `examples/` to a temporary directory
```
Comment (chrisguidry) [inline tests/cli/test_deployment_build.py:31]:
```suggestion
        """Uses a local file system method to copy `examples/` to a temporary directory
```

**Review by chrisguidry — APPROVED** [review]

File: src/prefect/testing/cli.py [hunk 2]
```diff
@@ -34,7 +35,7 @@ def check_contains(cli_result: Result, content: str, should_contain: bool):
     if should_contain:
         assert (
             content in output
-        ), f"Desired contents {display_content!r} not found in CLI ouput"
+        ), f"Desired contents {display_content!r} not found in CLI ouput\n{output!r}"
```
Comment (madkinsz) [inline src/prefect/testing/cli.py:38]: `invoke_and_assert` already displays all of the output by default on L75/84

**Review by bunchesofdonald — APPROVED** [review]

File: src/prefect/testing/cli.py (hunk 2 above)
Comment (cicdw) [reply src/prefect/testing/cli.py:38]: ah sorry, i wasn't seeing that for some reason - i can remove

Comment (cicdw) [reply src/prefect/testing/cli.py]: @madkinsz updated

**Pull request merged**